src/components/Api.js

```diff
@@ -0,0 +1,89 @@
+import { toggleSaving } from "../utlis/utils";
+
+export default class Api {
+  constructor(options) {
+    // constructor body
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+  _processResponse = (res) => {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  };
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/v1/cohort-3-en/cards`, {
+      method: "GET",
+      headers: this._headers,
+      //   headers: {
+      //     authorization: "35337f3b-35e8-4dc0-a9b5-b6c4dd4127c3",
+      //   },
+    }).then(this._processResponse);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/v1/cohort-3-en/users/me`, {
+      method: "GET",
+      headers: this._headers,
+    })
+      .then(this._processResponse)
+      .catch((err) => {
```

Comment:
Errors should be handled outside of the api class in `index.js`. Now `catch` block is placed in the middle of the chain, so the rest of the chain won't know anything about the error — but it should because we need to update the data only in case of success

The api call in `index.js` should look like this:

```js
// show loading
api.
  .then(...)
  .catch(...)
  .finally(() => {
    // hide loading
  })
```